pubsub.go

```diff
@@ -283,6 +296,7 @@ func (p *PubSub) processLoop(ctx context.Context) {
 
 			if p.blacklist.Contains(pid) {
 				log.Warning("ignoring connection from blacklisted peer: ", pid)
+				p.disc.NotifyNewPeer(pid)
```

Comment:
why notify for a blacklisted peer?

Reply:
We need to know if when a bootstrap attempt failed so we can remove them from the queue of peers we're waiting on. Blacklisting is a type of failure, we can tag the successes and failures differently if we want, but since this is all internal I don't know if there's really a rush to do this now,